sed/loader/flash/buffer_handler.py

```diff
@@ -214,6 +217,13 @@ def _fill_dataframes(self):
 
             self.df[typ] = df
         self.metadata.update({"file_statistics": file_stats, "filling": filling})
+        # Correct the 3-bit shift which encodes the detector ID in the 8s time
+        if self._config.get("split_sector_id_from_dld_time", False):
+            self.df["electron"], meta = split_dld_time_from_sector_id(
+                self.df["electron"],
+                config=self._config,
+            )
+            self.metadata.update(meta)
 
     def run(
```

Comment:
I'd suggest to make run return the two dataframes directly, then there is no need to access the class attributes. Maybe also rename to e.g. process_and_load_dataframe or whatever.